stakingcontract/sc.go

```diff
@@ -0,0 +1,194 @@
+package sc
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"math/big"
+
+	"github.com/urfave/cli"
+	"go.uber.org/zap"
+
+	"github.com/Evrynetlabs/evrynet-node/accounts/abi/bind"
+	"github.com/Evrynetlabs/evrynet-node/common"
+	stakingContracts "github.com/Evrynetlabs/evrynet-node/consensus/staking_contracts"
+	"github.com/Evrynetlabs/evrynet-node/core/types"
+	"github.com/Evrynetlabs/evrynet-node/crypto"
+	"github.com/Evrynetlabs/evrynet-node/evrclient"
+	"github.com/evrynet-official/evrynet-tools/lib/node"
+)
+
+var (
+	stakingScFlag = cli.StringFlag{
+		Name:  "stakingsc",
+		Usage: "address of staking smart-contract (Address in string format: 0x...)",
+		Value: "0x2d5bd25efa0ab97aaca4e888c5fbcb4866904e46",
+	}
+	senderPkFlag = cli.StringFlag{
+		Name:  "senderpk",
+		Usage: "the private key of admin/ owner/ user",
+		Value: "85af6fd1be0b4314fc00e8da30091541fff1a6a7159032ba9639fea4449e86cc",
+	}
+	candidateFlag = cli.StringFlag{
+		Name:  "candidate",
+		Usage: "the address of candidate (Address in string format: 0x...)",
+		Value: "0x71562b71999873DB5b286dF957af199Ec94617F7",
+	}
+	gasLimitFlag = cli.Uint64Flag{
+		Name:  "gaslimit",
+		Usage: "the gaslimit to execute the call to the contract.",
+		Value: 8000000,
+	}
+	amountFlag = cli.Int64Flag{
+		Name:  "amount",
+		Usage: "the amount.",
+		Value: 0,
+	}
+)
+
+// NewStakingFlag returns flags for Staking contract client
+func NewStakingFlag() []cli.Flag {
+	return []cli.Flag{stakingScFlag, senderPkFlag, candidateFlag, gasLimitFlag, amountFlag}
+}
+
+// ContractClient returns a struct
+type ContractClient struct {
+	Contract  *stakingContracts.StakingContracts
+	Client    *evrclient.Client
+	StakingSc common.Address
+	SenderPk  *ecdsa.PrivateKey
+	Candidate common.Address
+	Amount    *big.Int
+	TranOps   *bind.TransactOpts
+	Logger    *zap.SugaredLogger
+}
+
+// NewNewStakingFromFlags returns new instance of ContractClient
+func NewNewStakingFromFlags(ctx *cli.Context, logger *zap.SugaredLogger) (*ContractClient, error) {
+	var (
+		stakingSc      = ctx.String(stakingScFlag.Name)
+		senderPkString = ctx.String(senderPkFlag.Name)
+		candidate      = ctx.String(candidateFlag.Name)
+		amount         = new(big.Int).SetInt64(ctx.Int64(amountFlag.Name))
+		gasLimit       = ctx.Uint64(gasLimitFlag.Name)
+	)
+
+	if !common.IsHexAddress(stakingSc) {
+		return nil, errors.New("the address of staking sc is invalid")
+	}
+	if !common.IsHexAddress(candidate) {
+		return nil, errors.New("the address of candidate is invalid")
+	}
+	senderPk, err := crypto.HexToECDSA(senderPkString)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := node.NewEvrynetClientFromFlags(ctx)
+	if err != nil {
+		return nil, err
+	}
+	stakeSCAddr := common.HexToAddress(stakingSc)
+	contract, err := stakingContracts.NewStakingContracts(stakeSCAddr, client)
+	if err != nil {
+		return nil, err
+	}
+	optTrans := bind.NewKeyedTransactor(senderPk)
+	optTrans.GasLimit = gasLimit
```

Comment:
The comment was ignore previously. Please go through it. 
https://github.com/evrynet-official/evrynet-tools/pull/17#discussion_r392030886

Reply:
updated as request